app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package org.oppia.android.app.home.promotedlist
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.ViewModel
+import org.oppia.android.R
+import org.oppia.android.app.home.HomeItemViewModel
+import org.oppia.android.app.home.RouteToRecentlyPlayedListener
+import org.oppia.android.app.shim.IntentFactoryShim
+
+/** [ViewModel] promoted story list in [HomeFragment]. */
+class PromotedStoryListViewModel(
+  private val activity: AppCompatActivity,
+  private val internalProfileId: Int,
+  private val intentFactoryShim: IntentFactoryShim,
+  val promotedStoryList: List<PromotedStoryViewModel>
+) :
+  HomeItemViewModel(),
+  RouteToRecentlyPlayedListener {
+
+  /**
+   * Returns an [Int] for the padding placed at the start of the promoted stories list layout displayed on the
```

Comment:
Could just be 'Returns the padding...' since the type is included in the documentation "view" (e.g. if we actually generated HTML documentation pages from our KDocs the type of the function would be included from the function declaration). Another way to think about it: the documentation/API/contract is everything outside the curly braces, and the implementation is everything inside of the curly braces.

Ditto elsewhere in this PR.

Reply:
I like that perspective -- thanks, done